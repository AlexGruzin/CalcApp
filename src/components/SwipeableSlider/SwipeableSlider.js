import React, { useState, useEffect } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import SwipeableViews from 'react-swipeable-views';
import { autoPlay as autoPlayer, virtualize } from 'react-swipeable-views-utils';
import Slide from './Slide';

import './SwipeableSlider.styles.less';

const EnhancedSwipeableViews = autoPlayer(virtualize(SwipeableViews));

const mod = (index, overallAmount) => {
  const offset = index % overallAmount;
  return offset < 0 ? offset + overallAmount : offset;
};

const Slider = ({ className, slides, direction, autoPlay, onChange }) => {
  const [selectedIndex, setSelectedIndex] = useState(0);
  const [infiniteIndex, setInfiniteIndex] = useState(0);
  const [isFocused, setIsFocused] = useState(true);

  const handleFocusChange = () => {
    setIsFocused(true);
  };

  const handleBlurChange = () => {
    setIsFocused(false);
  };

  useEffect(() => {
    if (typeof window !== 'undefined') {
      window.addEventListener('focus', handleFocusChange);
      window.addEventListener('blur', handleBlurChange);
    }

    return () => {
      window.removeEventListener('focus', handleFocusChange);
      window.removeEventListener('blur', handleBlurChange);
    };
  }, []);

  const handleChangeSelectedIndex = index => {
    setSelectedIndex(mod(index, slides.length));
  };

  const handleChangeInfiniteIndex = index => {
    if (!isFocused) return;
    onChange(index);
    setInfiniteIndex(index);
    handleChangeSelectedIndex(index);
  };

  const manualChageSlideLeft = () => {
    if (!isFocused) return;
    onChange(infiniteIndex - 1);
    handleChangeInfiniteIndex(infiniteIndex - 1);
  };

  const manualChageSlideRight = () => {
    if (!isFocused) return;
    onChange(infiniteIndex + 1);
    handleChangeInfiniteIndex(infiniteIndex + 1);
  };

  const slideRenderer = params => {
    const { index, key } = params;
    const circularIndex = mod(index, slides.length);

    return (
      <Slide
        key={key}
        index={circularIndex}
        link={slides[circularIndex].link}
        image={slides[circularIndex].image}
        title={slides[circularIndex].title}
        description={slides[circularIndex].description}
      />
    );
  };

  return (
    <div className={classNames(className, 'slider')}>
      <EnhancedSwipeableViews
        springConfig={{
          duration: '0.6s',
          easeFunction: 'ease',
          delay: '0s',
        }}
        index={infiniteIndex}
        onChangeIndex={handleChangeInfiniteIndex}
        enableMouseEvents
        interval={7000}
        className="enhanced-swipeable-views"
        slideRenderer={slideRenderer}
        direction={direction}
        autoplay={autoPlay}
      />

      <div className="indicators-wrapper">
        <div tabIndex={0} role="button" className="slider__left-click" onClick={manualChageSlideLeft}>
          <img src="/svg/angle-left.svg" alt="angle-left" />
        </div>
        {slides.map((item, index) => {
          const statusClass = index === selectedIndex ? 'activated' : 'deactivated';
          return (
            <div key={index} className="indicator-button" onClick={() => handleChangeInfiniteIndex(index)}>
              <div className={classNames('indicator', statusClass)} />
            </div>
          );
        })}
        <div tabIndex={0} role="button" className="slider__right-click" onClick={manualChageSlideRight}>
          <img src="/svg/angle-left.svg" alt="angle-right" />
        </div>
      </div>
    </div>
  );
};

Slider.propTypes = {
  slides: PropTypes.array.isRequired,
  className: PropTypes.string,
  autoPlay: PropTypes.bool,
  direction: PropTypes.string,
  onChange: PropTypes.func,
};

Slider.defaultProps = {
  className: '',
  direction: 'incremental', // decremental
  autoPlay: false,
  onChange: () => undefined,
};

export default Slider;
