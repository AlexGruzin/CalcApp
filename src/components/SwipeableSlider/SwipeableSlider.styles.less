.slider {
  display: flex;
  flex-direction: column;
  width: 570px;
  height: calc(408px + 36px);
  justify-content: center;
}

.enhanced-swipeable-views {
  justify-content: center;
  height: 408px;
  width: 100%;
}

.level-1 {
  left: -100%;
}
.level0 {
  left: 0;
}
.level1 {
  left: 100%;
}

.indicators-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 24px;
}

.indicator {
  cursor: pointer;
  transition: all 0.5s ease;
  display: flex;
  border-radius: 50%;
  width: var(--space-x3);
  height: var(--space-x3);
  margin: auto var(--space-x3);
  border-style: solid;
  border-width: 1px;
}

.indicator-button {
  display: flex;
  height: 100%;
  user-select: none;
  cursor: pointer;
}

.indicator.activated {
  background: var(--white);
  border-color: var(--white);
}

.indicator.deactivated {
  background: transparent;
  border-color: var(--grey300);
}

.slider__left-click,.slider__right-click {
  cursor: pointer;
  height: 20px;
  width: 20px;
  margin: 0 var(--space-x3);
}

.slider__right-click {
  transform: rotate(180deg);
}
