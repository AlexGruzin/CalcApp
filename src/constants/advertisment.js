export const ADVERT = {
  CORDIAL: 'CORDIAL',
  FACEBOOK: 'FACEBOOK',
  GOOGLE_ADS: 'GOOGLE_ADS',
  GOOGlE_ANALYTICS: 'GOOGLE_ANALYTICS',
  INSTAGRAM: 'INSTAGRAM',
  KLAVIYO: 'KLAVIYO',
  TIKTOK: 'TIKTOK',
  TWITTER: 'TWITTER',
  SHOPIFY: 'SHOPIFY',
  SHOPIFY_SIGNUP: 'SHOPIFY_SIGNUP',
};

export const SOCIAL = {
  ENVELOPE: 'ENVELOPE',
  BITMAP: 'BITMAP',
};

export const ADVERT_ICONS = {
  [ADVERT.FACEBOOK]: 'facebook.png',
  [ADVERT.INSTAGRAM]: 'instagram.png',
  [SOCIAL.ENVELOPE]: 'envelope.png',
  [SOCIAL.BITMAP]: 'bitmap.png',
};
