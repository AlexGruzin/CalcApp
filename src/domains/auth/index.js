import AllActionsTypes from './actionTypes';
import * as AllActionsCreators from './actions';
import saga from './saga';
import reducer from './reducer';
import * as selectors from './selectors';

export { AllActionsTypes as ActionsTypes, AllActionsCreators as ActionsCreators, saga, reducer, selectors };
