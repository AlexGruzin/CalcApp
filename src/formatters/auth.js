export const formatUserInfo = data => {
  const {
    account_non_expired,
    account_non_locked,
    credentials_non_expired,
    enabled,
    authorities,
    user_id,
    username,
  } = data;

  return {
    accountNonExpired: account_non_expired,
    accountNonLocked: account_non_locked,
    credentialsNonExpired: credentials_non_expired,
    enabled,
    authorities,
    userId: user_id,
    username,
  };
};
