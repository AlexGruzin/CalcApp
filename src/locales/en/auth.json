{
  "logInForm": {
    "newUser": "Are you a new user?"
  },
  "signUpForm": {
    "haveAccount": "Already have an account?",
    "oneStep": "One step left",
    "visitMail": "Visit your mail:",
    "complete": "to complete registration."
  },

  "signUp": "Sign Up",
  "logIn": "Log in",
  "password": "Password",
  "username": "Username",
  "emailAddress": "Email address",
  "createAccount": "Create your Calc account:",
  "forgot": "Forgot your password?",
  "welcome": "Welcome",
  "description": "Log into your Calc account:",
  "passwordReset": "Password reset",
  "termsDescriptionPartOne": "By clicking Log in, you agree to our",
  "termsDescriptionPartOneSign": "By clicking Create Account, you agree to our",
  "termsDescriptionPartTwo": "and have read and acknowledge our",
  "terms": "Terms",
  "privacyStatement": "Privacy Statement",
  "learnMore": "Learn More",
  "back": "Back"
}
