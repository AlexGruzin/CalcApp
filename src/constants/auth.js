import * as Routing from 'constants/routing';

export const HEADER_FREE_PAGES = [
  Routing.LOG_IN,
  Routing.SIGN_UP,
  Routing.RESET_PASSWORD,
  Routing.CONFIRM_OAUTH,
  Routing.CONFIRM_SIGN_UP,
  Routing.CONFIRM_EMAIL_CHANGE,
  Routing.CONFIRM_USER,
  Routing.CONFIRM_PASSWORD,
  Routing.SHOPIFY_SIGNUP,
  Routing.CONFIRM_SHOPIFY_SIGNUP,
  Routing.SURVEY,
  Routing.TERMS_LINK,
  Routing.PRIVACY_STATEMENT_LINK,
  Routing.WELCOME,
];

export const DELAY_TO_SEE_AND_READ_REQUEST_STATUS = 1000;

export const DEVELOPMENT_CASES_DELAY = 10000;

const temp_title = 'We’ll help you plan and show you how to get there.';
const temp_description =
  'By connecting with your Storefront and Analytics, we can help you create a plan for the next year — And will give you the steps to get there.';

export const SLIDES_DATA = [
  {
    image: 'slide1.png',
    link: 'slide1',
    title: temp_title,
    description: temp_description,
  },
  {
    image: 'slide1.png',
    link: 'slide2',
    title: temp_title,
    description: temp_description,
  },
  {
    image: 'slide1.png',
    link: 'slide3',
    title: temp_title,
    description: temp_description,
  },
];

export const CONFIRMATION_STATUS = {
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
