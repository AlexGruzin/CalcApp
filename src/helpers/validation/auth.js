import { CODES } from 'constants/request';

const EMAIL_REGEX = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

export const emailValidation = () => ({
  pattern: {
    value: EMAIL_REGEX,
    message: 'The email you entered is invalid.',
  },
  required: {
    value: true,
    message: 'Email is required.', // now this one is overridden with native html behaviour
  },
});

export const passwordValidation = () => ({
  required: true,
  minLength: {
    value: 8,
    message: 'Password is too short.',
  },
});

export const getErrorMsgByResponse = data => {
  switch (data.status) {
    case CODES.INTERNAL_SERVER_ERROR: {
      return 'Internal server error.';
    }
    default: {
      return data?.response?.message || 'app:error.unknown';
    }
  }
};
