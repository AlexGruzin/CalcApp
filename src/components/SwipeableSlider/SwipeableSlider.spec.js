import React from 'react';
import { waitFor, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SLIDES_DATA } from 'constants/auth';

import SwipeableSlider from './SwipeableSlider';

describe('SwipeableSlider tests', () => {
  it('Check snapshots of SwipeableSlider are equal.', () => {
    const { asFragment } = render(<SwipeableSlider slides={SLIDES_DATA} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Check SwipeableSlider click right change slide.', () => {
    const mockOnChange = jest.fn();
    const { getAllByRole } = render(<SwipeableSlider onChange={mockOnChange} slides={SLIDES_DATA} />);
    const btnArrow = getAllByRole('button');
    userEvent.click(btnArrow[1]);

    expect(mockOnChange).toBeCalledWith(1);
  });

  it('Check SwipeableSlider click left change slide.', () => {
    const mockOnChange = jest.fn();
    const { getAllByRole } = render(<SwipeableSlider onChange={mockOnChange} slides={SLIDES_DATA} />);
    const btnArrow = getAllByRole('button');

    userEvent.click(btnArrow[0]);
    expect(mockOnChange).toBeCalledWith(-1);
  });

  it('Check SwipeableSlider click indicator change slide.', () => {
    const mockOnChange = jest.fn();
    const { container } = render(<SwipeableSlider onChange={mockOnChange} slides={SLIDES_DATA} />);
    const btnIndicator = container.querySelectorAll('.indicator-button');

    userEvent.click(btnIndicator[2]);
    expect(mockOnChange).toBeCalledWith(2);
  });
});
