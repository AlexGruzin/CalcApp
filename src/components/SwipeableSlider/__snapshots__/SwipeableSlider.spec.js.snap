// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableSlider tests Check snapshots of SwipeableSlider are equal. 1`] = `
<DocumentFragment>
  <div
    class="slider"
  >
    <div
      class="enhanced-swipeable-views"
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
      >
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="slide"
          >
            <div
              class="slide__image"
              style="background-image: url(/images/slide1.png);"
            />
            <h4
              class="slide__title"
            >
              We’ll help you plan and show you how to get there.
            </h4>
            <p
              class="slide__description"
            >
              By connecting with your Storefront and Analytics, we can help you create a plan for the next year — And will give you the steps to get there.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="indicators-wrapper"
    >
      <div
        class="slider__left-click"
        role="button"
        tabindex="0"
      >
        <img
          alt="angle-left"
          src="/svg/angle-left.svg"
        />
      </div>
      <div
        class="indicator-button"
      >
        <div
          class="indicator activated"
        />
      </div>
      <div
        class="indicator-button"
      >
        <div
          class="indicator deactivated"
        />
      </div>
      <div
        class="indicator-button"
      >
        <div
          class="indicator deactivated"
        />
      </div>
      <div
        class="slider__right-click"
        role="button"
        tabindex="0"
      >
        <img
          alt="angle-right"
          src="/svg/angle-left.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
